drivers/flash/spi_nor.c

```diff
@@ -1097,7 +1097,11 @@ static int spi_nor_process_bfp(const struct device *dev,
 	struct jesd216_erase_type *etp = data->erase_types;
 	const size_t flash_size = jesd216_bfp_density(bfp) / 8U;
 
-	LOG_INF("%s: %u MiBy flash", dev->name, (uint32_t)(flash_size >> 20));
+	if (flash_size < (1024U * 1024U)) {
+		LOG_INF("%s: %u kiBy flash", dev->name, (uint32_t)(flash_size >> 10));
```

Comment:
Would you consider replacing `k` with `%c` in the format string and setting it to "M" or "k", depending on the result of comparison?

Reply:
I will and is working on it. I just made a mistake that I will fix.

Reply:
OK, I fixed it. Hope it is OK.